### PR TITLE
[CSUB-800] Testnet release 3.12.0

### PR DESCRIPTION
**Pull request by pLabarta**
# Description of proposed changes

Update testnet branch to prepare first testnet release

---

Practical tips for PR review & merge:

- [ ] All GitHub Actions report PASS
- [ ] Newly added code/functions have unit tests
  - [ ] Coverage tools report all newly added lines as covered
  - [ ] The positive scenario is exercised
  - [ ] Negative scenarios are exercised, e.g. assert on all possible errors
  - [ ] Assert on events triggered if applicable
  - [ ] Assert on changes made to storage if applicable
- [ ] Modified behavior/functions - try to make sure above test items are covered
- [ ] Integration tests are added if applicable/needed

### CODE REVIEW

**Review by zacharyfrederick — APPROVED** [review]

**Review by zacharyfrederick — CHANGES_REQUESTED** [review]
Reviewing locally

**Review by zacharyfrederick — APPROVED** [review]

**Review by nathanwhit — APPROVED** [review]
Left some comments. Nothing blocking, but stuff to maybe address in the future.

File: .github/authorized_keys [hunk 1]
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDOYiBGthrSNqUJdN9h9PHzXQL8cP8gj5pP9LZDx7BVgt/Knm9NwAe9hD/7fs9zmyECmZ5ubHDqG0x7Hb7DAjl+oPkCOxqRj5Npfvl1VRwwgXl3ymfI3JJpF7Cna4n0XdylBsTiwOL1/zoVXEJgTYEDEsP4gv65i8M/uWlsrfFwHLDEr3EQKnA0H4Ekz1CU2n9MFprX1hzA5IItozQUsYxKTPr1mxNTi1AFMhDEztMelvPO1OuC8MBZURR9S/+SlZ8ydCUcwl0gdCcgpUfouiuN9Yr9UbiV/yrAxEY3oKX8OegFmWEioUIZoFSiXl3sNP39ntOR4i+GV54g4omN6JbN3ios9LXBPMuvCy1NgFYPmDmmSEuo7n2IsP3pcXjZXpl4Ymwvn4RJviOZq8vghF5p0YMtCis1LxHL90epibxGoV6nc5isrOzqfgNUA9IS1zx7ZMoaF1PT8QcT8NzgNocrrhH4xGeTLD7WOOxykVASBnIXea8p7dl29/G0ThmS7Bc= atodorov@macbook.pro
```
[the remark this answers is not in the document]
Comment (atodorov) [reply .github/authorized_keys:1]: Allows you to ssh into the self-hosted runner VM. This is my public ssh key.

File: Dockerfile [hunk 2]
```diff
@@ -1,36 +1,54 @@
...
 
-# The execution will re-compile the project to run it
-# This allows to modify the code and not have to re-compile the
-# dependencies.
-CMD cargo run --bin frontier-template-node "--$PROFILE" -- --dev
+FROM runtime-base
+EXPOSE 30333/tcp
+EXPOSE 30333/udp
+EXPOSE 9944 9933 9615
+ENTRYPOINT [ "/bin/creditcoin-node" ]
```
[the remark this answers is not in the document]
Comment (atodorov) [reply Dockerfile:45]: +1 for keeping the name consistent everywhere. I think this is a left-over from when we weren't sure how to call it.

https://github.com/gluwa/creditcoin3/pull/160

**Pull request merged**